File: backend/userApi/Application/Services/UserService.cs
using System.Collections.Generic;
using System.Linq;
using BCrypt.Net;
// IMPORTS CORRETOS:
using UserApi.API.DTOs;
using UserApi.Domain.Entities;
using UserApi.Domain.Interfaces;
using UserApi.Application.Interfaces;

namespace UserApi.Application.Services
{
    public class UserService : IUserService
    {
        private readonly IUserRepository _userRepository;

        public UserService(IUserRepository userRepository)
        {
            _userRepository = userRepository;
        }

        public UserDTO RegisterUser(UserDTO userDto)
        {
            var hashedPassword = BCrypt.Net.BCrypt.HashPassword(userDto.Password);

            var user = new User
            {
                Name = userDto.Name,
                Email = userDto.Email,
                Phone = userDto.Phone,
                Address = userDto.Address,
                Password = hashedPassword,
                Photo = userDto.Photo,
                Role = "user" // Segurança: força user comum
            };
            _userRepository.Add(user);

            return new UserDTO
            {
                Id = user.Id,
                Name = user.Name,
                Email = user.Email,
                Role = user.Role,
                Phone = user.Phone,
                Address = user.Address,
                Password = user.Password,
                Photo = user.Photo
            };
        }

        public UserDTO? GetUserDetails(int id)
        {
            var user = _userRepository.GetById(id);
            return user != null ? new UserDTO
            {
                Id = user.Id,
                Name = user.Name,
                Email = user.Email,
                Role = user.Role,
                Phone = user.Phone,
                Address = user.Address,
                Photo = user.Photo
            } : null;
        }

        public List<UserDTO> GetAllUsers()
        {
            return _userRepository.GetAll().Select(user => new UserDTO
            {
                Id = user.Id,
                Name = user.Name,
                Email = user.Email,
                Role = user.Role,
                Phone = user.Phone,
                Address = user.Address,
                Password = user.Password,
                Photo = user.Photo
            }).ToList();
        }

        public UserDTO? UpdateUser(int id, UserDTO userDto)
        {
            var user = _userRepository.GetById(id);
            if (user == null) return null;

            user.Name = userDto.Name;
            user.Email = userDto.Email;
            user.Phone = userDto.Phone;
            user.Address = userDto.Address;

            if (!string.IsNullOrEmpty(userDto.Photo))
                user.Photo = userDto.Photo;

            if (!string.IsNullOrEmpty(userDto.Password))
                user.Password = BCrypt.Net.BCrypt.HashPassword(userDto.Password);

            _userRepository.Update(user);

            return new UserDTO
            {
                Id = user.Id,
                Name = user.Name,
                Email = user.Email,
                Role = user.Role,
                Phone = user.Phone,
                Address = user.Address,
                Password = "", 
                Photo = user.Photo
            };
        }

        public bool DeleteUser(int id)
        {
            var user = _userRepository.GetById(id);
            if (user == null) return false;
            _userRepository.Delete(id);
            return true;
        }

        public UserDTO? ValidateUser(string email, string password)
        {
            var user = _userRepository.GetByEmail(email);
            if (user == null || !BCrypt.Net.BCrypt.Verify(password, user.Password))
                return null;

            return new UserDTO
            {
                Id = user.Id,
                Name = user.Name,
                Email = user.Email,
                Role = user.Role,
                Phone = user.Phone,
                Address = user.Address,
                Photo = user.Photo
            };
        }
    }
}

File: frontend/src/services/apis/userService.ts
import axios from 'axios';
import { userAPI } from './api';

export interface User {
  id: number;
  name: string;
  email: string;
  password?: string;
  photo: string;
  phone: string;
  address: string;
}

const authHeader = () => {
  const token = typeof window !== 'undefined' ? localStorage.getItem('token') : null;
  return token ? { Authorization: `Bearer ${token}` } : {};
};

const userService = {
  getAll: async (): Promise<User[]> => {
    const response = await axios.get(userAPI.getAll(), {
      headers: authHeader(),
    });
    return response.data;
  },

  getById: async (id: number): Promise<User> => {
    const response = await axios.get(userAPI.getById(id), {
      headers: authHeader(),
    });
    return response.data;
  },

  create: async (user: Omit<User, 'id'>) => {
    const response = await axios.post(userAPI.create(), user, {
      headers: authHeader(),
    });
    return response.data;
  },

  update: async (id: number, user: Partial<User>) => {
    const response = await axios.put(userAPI.update(id), user, {
      headers: authHeader(),
    });
    return response.data;
  },

  delete: async (id: number) => {
    const response = await axios.delete(userAPI.delete(id), {
      headers: authHeader(),
    });
    return response.data;
  },
};

export default userService;
